### PR TITLE
Intercept access to interface members known to be not-supported by the target runtime.

### PR DESCRIPTION
**Pull request by AlekseyTs**
@dotnet/roslyn-compiler Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
Retest windows_debug_vs-integration_prtest please

**Comment by AlekseyTs** [comment]
Retest windows_release_vs-integration_prtest please

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs [hunk 1]
```diff
@@ -30649,5 +30659,213 @@ static void Main()
...
+        }
+
+        [Fact]
+        public void EntryPoint_01()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    static void Main()
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: Thanks ;-)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

File: src/Compilers/CSharp/Portable/CSharpResources.resx [hunk 2]
```diff
@@ -2510,7 +2510,7 @@ A catch() block after a catch (System.Exception e) block can catch non-CLS excep
     <value>Could not find '{0}' specified for Main method</value>
   </data>
   <data name="ERR_MainClassNotClass" xml:space="preserve">
-    <value>'{0}' specified for Main method must be a valid non-generic class or struct</value>
+    <value>'{0}' specified for Main method must be a valid non-generic class or struct or interface</value>
```
Comment (agocke) [inline src/Compilers/CSharp/Portable/CSharpResources.resx]: Perhaps `non-generic class, struct, or interface`?

Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/CSharpResources.resx]: > Perhaps non-generic class, struct, or interface?

Sure, I'll update the message in the next PR. Thanks.

**Pull request merged**